lemur/certificates/service.py

```diff
@@ -560,18 +560,35 @@ def query_common_name(common_name, args):
     :return:
     """
     owner = args.pop("owner")
-    if not owner:
-        owner = "%"
-
     # only not expired certificates
     current_time = arrow.utcnow()
 
-    result = (
-        Certificate.query.filter(Certificate.cn.ilike(common_name))
-        .filter(Certificate.owner.ilike(owner))
-        .filter(Certificate.not_after >= current_time.format("YYYY-MM-DD"))
-        .all()
-    )
+    if common_name == "%" and not owner:
+        result = (
+            Certificate.query.filter(Certificate.not_after >= current_time.format("YYYY-MM-DD"))
+            .filter(not_(Certificate.revoked))
+            .filter(not_(Certificate.replaced.any()))  # ignore rotated certificates to avoid duplicates
+            .all()
+        )
+    elif common_name == "%":
+        # all valid certs from the owner
+        result = (
+            Certificate.query.filter(Certificate.not_after >= current_time.format("YYYY-MM-DD"))
+            .filter(Certificate.owner == owner)
```

Comment:
Is it possible that a request might come in where owner = '%foobar%'?  If so, then this request will likely fail.

Maybe something like the following?
if "%" in owner:
  Query using Certificate.owner.ilike(owner)
else:
  Query using Certificate.owner == owner